The function "IsWindows" on type "System.OperatingSystem" is not available for execution as an MSBuild property function.

### Package

Sentry

### .NET Flavor

.NET Framework

### .NET Version

4.8.2

### OS

Windows

### OS Version

Windows Server 2022

### Development Environment

Visual Studio v17.x

### SDK Version

5.6.0

### Self-Hosted Sentry Version

_No response_

### Workload Versions

-

### UseSentry or SentrySdk.Init call

-

### Steps to Reproduce

Azure DevOps Pipeline Build

### Expected Result

-

### Actual Result

##[error]packages\Sentry.5.6.0\build\Sentry.targets(18,14): Error MSB4185: The function "IsWindows" on type "System.OperatingSystem" is not available for execution as an MSBuild property function.

This is the build log:

```
Starting: Build solution CleverMatch.sln
==============================================================================
Task         : Visual Studio build
Description  : Build with MSBuild and set the Visual Studio version property
Version      : 1.253.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/build/visual-studio-build
==============================================================================
"C:\agent\_work\_tasks\VSBuild_71a9a2d3-a98a-4caa-96ab-affca411ecda\1.253.0\node_modules\azure-pipelines-tasks-msbuildhelpers\tools\vswhere.exe" -version [17.0,18.0) -latest -format json
"C:\agent\_work\_tasks\VSBuild_71a9a2d3-a98a-4caa-96ab-affca411ecda\1.253.0\node_modules\azure-pipelines-tasks-msbuildhelpers\tools\vswhere.exe" -version [17.0,18.0) -products Microsoft.VisualStudio.Product.BuildTools -latest -format json
"C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Current\Bin\amd64\msbuild.exe" "C:\agent\_work\2\s\CleverMatch.sln" /nologo /nr:false /t:"Clean" /dl:CentralLogger,"C:\agent\_work\_tasks\VSBuild_71a9a2d3-a98a-4caa-96ab-affca411ecda\1.253.0\node_modules\azure-pipelines-tasks-msbuildhelpers\tools\Microsoft.TeamFoundation.DistributedTask.MSBuild.Logger.dll";"RootDetailId=0d43ebe9-1c7a-41e4-8573-3f813575300b|SolutionDir=C:\agent\_work\2\s|enableOrphanedProjectsLogs=true"*ForwardingLogger,"C:\agent\_work\_tasks\VSBuild_71a9a2d3-a98a-4caa-96ab-affca411ecda\1.253.0\node_modules\azure-pipelines-tasks-msbuildhelpers\tools\Microsoft.TeamFoundation.DistributedTask.MSBuild.Logger.dll" /p:DeployOnBuild=true /p:WebPublishMethod=Package /p:PackageAsSingleFile=true /p:SkipInvalidConfigurations=true /p:PackageLocation="C:\agent\_work\2\a\\" /p:PrecompileBeforePublish=true /p:DestinationType=AzureWebSite /p:Platform="AnyCPU" /p:Configuration=Release /p:UseMerge=true /p:SingleAssemblyName=AppCode /p:Verbosity=Normal /p:platform="any cpu" /p:configuration="release" /p:VisualStudioVersion="17.0" /p:_MSDeployUserAgent="VSTS_10c24b9e-63be-411e-bf4b-764899d4df5a_build_2_0"
Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.
Build started 4/25/2025 4:48:15 PM.

Project "C:\agent\_work\2\s\CleverMatch.sln" on node 1 (Clean target(s)).
ValidateSolutionConfiguration:
  Building solution configuration "release|any cpu".
ValidateProjects:
  The project "CleverMatch.UnitTests" is not selected for building in solution configuration "Release|Any CPU".
  The project "Facebook.ApiClient.Test" is not selected for building in solution configuration "Release|Any CPU".
##[error]packages\Sentry.5.6.0\build\Sentry.targets(18,14): Error MSB4185: The function "IsWindows" on type "System.OperatingSystem" is not available for execution as an MSBuild property function.
Project "C:\agent\_work\2\s\CleverMatch.sln" (1) is building "C:\agent\_work\2\s\CleverMatch.WebJobs.HostBuilder\CleverMatch.WebJobs.HostBuilder.csproj.metaproj" (2) on node 1 (Clean target(s)).
Project "C:\agent\_work\2\s\CleverMatch.WebJobs.HostBuilder\CleverMatch.WebJobs.HostBuilder.csproj.metaproj" (2) is building "C:\agent\_work\2\s\CleverMatch.Core\CleverMatch.Core.csproj" (3) on node 1 (Clean target(s)).
C:\agent\_work\2\s\packages\Sentry.5.6.0\build\Sentry.targets(18,14): error MSB4185: The function "IsWindows" on type "System.OperatingSystem" is not available for execution as an MSBuild property function. [C:\agent\_work\2\s\CleverMatch.Core\CleverMatch.Core.csproj]
Done Building Project "C:\agent\_work\2\s\CleverMatch.Core\CleverMatch.Core.csproj" (Clean target(s)) -- FAILED.
Done Building Project "C:\agent\_work\2\s\CleverMatch.WebJobs.HostBuilder\CleverMatch.WebJobs.HostBuilder.csproj.metaproj" (Clean target(s)) -- FAILED.
##[error]packages\Sentry.5.6.0\build\Sentry.targets(18,14): Error MSB4185: The function "IsWindows" on type "System.OperatingSystem" is not available for execution as an MSBuild property function.
Project "C:\agent\_work\2\s\CleverMatch.sln" (1) is building "C:\agent\_work\2\s\CleverMatch.WebJobs.MarketingAutomation\CleverMatch.WebJobs.MarketingAutomation.csproj" (4) on node 1 (Clean target(s)).
C:\agent\_work\2\s\packages\Sentry.5.6.0\build\Sentry.targets(18,14): error MSB4185: The function "IsWindows" on type "System.OperatingSystem" is not available for execution as an MSBuild property function. [C:\agent\_work\2\s\CleverMatch.WebJobs.MarketingAutomation\CleverMatch.WebJobs.MarketingAutomation.csproj]
Done Building Project "C:\agent\_work\2\s\CleverMatch.WebJobs.MarketingAutomation\CleverMatch.WebJobs.MarketingAutomation.csproj" (Clean target(s)) -- FAILED.

```

Build (x64) command parameter are:

`/p:DeployOnBuild=true /p:WebPublishMethod=Package /p:PackageAsSingleFile=true /p:SkipInvalidConfigurations=true /p:PackageLocation="$(build.artifactstagingdirectory)\\" /p:PrecompileBeforePublish=true /p:DestinationType=AzureWebSite /p:Platform="AnyCPU" /p:Configuration=Release /p:UseMerge=true /p:SingleAssemblyName=AppCode /p:Verbosity=Normal`

YAML:

```
steps:
- task: VSBuild@1
  displayName: 'Build solution CleverMatch.sln'
  inputs:
    solution: CleverMatch.sln
    vsVersion: 17.0
    msbuildArgs: '/p:DeployOnBuild=true /p:WebPublishMethod=Package /p:PackageAsSingleFile=true /p:SkipInvalidConfigurations=true /p:PackageLocation="$(build.artifactstagingdirectory)\\" /p:PrecompileBeforePublish=true /p:DestinationType=AzureWebSite /p:Platform="AnyCPU" /p:Configuration=Release /p:UseMerge=true /p:SingleAssemblyName=AppCode /p:Verbosity=Normal'
    platform: '$(BuildPlatform)'
    configuration: '$(BuildConfiguration)'
    clean: true
    msbuildArchitecture: x64
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,7 @@
 ### Fixes
 
 - Work around iOS SHA1 bug ([#4143](https://github.com/getsentry/sentry-dotnet/pull/4143))
+- Fixes build error when building .NET Framwork applications using Sentry 5.6.0: `MSB4185 :The function "IsWindows" on type "System.OperatingSystem" is not available` ([#4160](https://github.com/getsentry/sentry-dotnet/pull/4160))
 
 ### Dependencies
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,7 +9,7 @@
 ### Fixes
 
 - Work around iOS SHA1 bug ([#4143](https://github.com/getsentry/sentry-dotnet/pull/4143))
-- Fixes build error when building .NET Framwork applications using Sentry 5.6.0: `MSB4185 :The function "IsWindows" on type "System.OperatingSystem" is not available` ([#4160](https://github.com/getsentry/sentry-dotnet/pull/4160))
+- Fixes build error when building .NET Framework applications using Sentry 5.6.0: `MSB4185 :The function "IsWindows" on type "System.OperatingSystem" is not available` ([#4160](https://github.com/getsentry/sentry-dotnet/pull/4160))
 
 ### Dependencies
 

--- a/src/Sentry/buildTransitive/Sentry.targets
+++ b/src/Sentry/buildTransitive/Sentry.targets
@@ -15,7 +15,7 @@
     <Error Text="Android projects using Sentry cannot build using AndroidEnableAssemblyCompression = false due to a Microsoft issue.  Please follow https://github.com/dotnet/android/issues/9752" />
   </Target>
 
-  <ItemGroup Condition="$([System.OperatingSystem]::IsWindows())">
+  <ItemGroup Condition="$([MSBuild]::IsOSPlatform('Windows'))">
     <!-- See: https://github.com/getsentry/sentry-dotnet/pull/4111 -->
     <LinkerArg Include="/NODEFAULTLIB:MSVCRT" />
   </ItemGroup>

